ast/ast.fbs

```diff
@@ -0,0 +1,359 @@
+namespace fbast;
+
+union Statement {
+  BadStatement,
+  VariableAssignment,
+  MemberAssignment,
+  ExpressionStatement,
+  ReturnStatement,
+  OptionStatement,
+  BuiltinStatement,
+  TestStatement,
+}
+
+union Assignment {
+  MemberAssignment,
+  VariableAssignment,
+}
+
+union Expression {
+  StringExpression,
+  ParenExpression,
+  ArrayExpression,
+  FunctionExpression,
+  BinaryExpression,
+  BooleanLiteral,
+  CallExpression,
+  ConditionalExpression,
+  DateTimeLiteral,
+  DurationLiteral,
+  FloatLiteral,
+  Identifier,
+  IntegerLiteral,
+  LogicalExpression,
+  MemberExpression,
+  IndexExpression,
+  ObjectExpression,
+  PipeExpression,
+  PipeLiteral,
+  RegexpLiteral,
+  StringLiteral,
+  UnaryExpression,
+  UnsignedIntegerLiteral,
+}
+
+struct Position {
+  line:int;
+  column:int;
+}
+
+table SourceLocation {
+  file:string;
+  start:Position;
+  end:Position;
+  source:string;
+}
+
+table BaseNode {
+  loc:SourceLocation;
+  errors:[string];
+}
+
+table Package {
+  base_node:BaseNode;
+  path:string;
+  package:string;
+  files:[File];
+}
+
+table File {
+  base_node:BaseNode;
+  name:string;
+  package:PackageClause;
+  imports:[ImportDeclaration];
+  body:[WrappedStatement];
+}
+
+table PackageClause {
+  base_node:BaseNode;
+  name:Identifier;
+}
+
+table ImportDeclaration {
+  base_node:BaseNode;
+  as:Identifier;
+  path:StringLiteral;
+}
+
+// Statements
+
+// Unions cannot be vector elements,
+// so wrap the union in a table.
+table WrappedStatement {
+  statement:Statement;
+}
+
+table BadStatement {
+  base_node:BaseNode;
+  text:string;
+}
+
+table VariableAssignment {
+  base_node:BaseNode;
+  id:Identifier;
+  // Calling the below "init" creates errors in generated Go code
+  init_:Expression;
+}
+
+table MemberAssignment {
+  base_node:BaseNode;
+  member:MemberExpression;
+  // Calling the below "init" creates errors in generated Go code
+  init_:Expression;
+}
+
+table ExpressionStatement {
+  base_node:BaseNode;
+  expression:Expression;
+}
+
+table ReturnStatement {
+  base_node:BaseNode;
+  argument:Expression;
+}
+
+table OptionStatement {
+  base_node:BaseNode;
+  assignment:Assignment;
+}
+
+table BuiltinStatement {
+  base_node:BaseNode;
+  id:Identifier;
+}
+
+table TestStatement {
+  base_node:BaseNode;
+  assignment:Assignment;
+}
+
+// Expressions
+
+// Unions cannot appear in vectors or other unions,
+// so provide a wrapped version where needed.
+table WrappedExpression {
+  expr:Expression;
+}
+
+enum OperatorKind : byte {
```

Comment:
In the Rust code I removed the `Kind` suffix from all of these types since they were not needed. Are they needed here or can we shorten this to `Operator`?

Reply:
Yes, it can be shortened.